### PR TITLE
2579 add helpertext support for chipset

### PR DESCRIPTION
**Pull request by Kiarokh**
fix https://github.com/Lundalogik/crm-feature/issues/2579

## Review:
- [x] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [x] Commits have the correct *type* for the changes made
- [x] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [x] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/chip-set/chip-set.tsx [hunk 1]
```diff
@@ -62,6 +62,15 @@ export class ChipSet {
     @Prop({ reflect: true })
     public label: string;
 
+    /**
+     * Optional helper text to display below the chipset.
+     * When type is not `input`, the helper text is always displayed
+     * if the device is touch screen; otherwise it is shown when chip-set
+     * is hovered.
```
Comment (adrianschmidt) [inline src/components/chip-set/chip-set.tsx:69]: And when the type _is_ `input`?

Comment (Kiarokh) [reply src/components/chip-set/chip-set.tsx:69]: > otherwise it is shown when chip-set is hovered.

Maybe it wasn't so clear after all 😄 

It's actually a bit incomplete too. Because that'll be shown when chipset is hovered or focused with keyboard navigation. I'll add a fixup.

thanks for the feedback.

Comment (Kiarokh) [reply src/components/chip-set/chip-set.tsx:69]: Thanks for the feedback. It was unclear.. I'll add a fixup soon

Comment (adrianschmidt) [reply src/components/chip-set/chip-set.tsx:69]: I assume that when the type is input, the helper text just works the way it does for a limel-input-field, right? If so, I think it's ok to just write that. This comment doesn't have to describe that behaviour. It just needs to not seem like a part of the comment is missing, because that will confuse people 😄

Comment (Kiarokh) [reply src/components/chip-set/chip-set.tsx]: I was too quick in responding. 😄 changed my answer later, but I think you already saw it. 

I see what you mean. I've pushed a fixup now. Do you think I should simplify the comment and instead make an example? Or is it OK as it is?

File: src/components/chip-set/chip-set.tsx [hunk 2]
```diff
@@ -62,6 +62,17 @@ export class ChipSet {
     @Prop({ reflect: true })
     public label: string;
 
+    /**
+     * Optional helper text to display below the chipset.
+     * When type is not `input`, the helper text is displayed below the
+     * input field when it has focus.
+     * When type is not `input`, the helper text is always displayed
```
Comment (adrianschmidt) [inline src/components/chip-set/chip-set.tsx:69]: Both of these say "When type is not \`input\`" now 😄

I commented about saying something like "works just like for limel-input-field", but that was because I assumed we simply passed the helper-text value on to limel-input-field when type is `input`, but that doesn't seem to be the case?

Is the mdc-chip-set component handling the showing and hiding of the helper text, or are we doing that?

Comment (Kiarokh) [reply src/components/chip-set/chip-set.tsx:69]: > Both of these say "When type is not `input`" now 😄

Oh lord 🙈 my memory…

> Is the mdc-chip-set component handling the showing and hiding of the helper text, or are we doing that?

yes, for the type=input, it is MDC which is doing it. I didn't do anything special. Just added the helper line. For other types, I did some CSS rules for touchscreen and focus and hover.

Comment (Kiarokh) [reply src/components/chip-set/chip-set.tsx:69]: for input field, we say:
````tsx
    /**
     * Optional helper text to display below the input field when it has focus
     */
````

Comment (adrianschmidt) [reply src/components/chip-set/chip-set.tsx]: Yeah, ok, then I think it's good as it is now. Nice work!

File: src/components/chip-set/chip-set.tsx [hunk 3]
```diff
@@ -276,6 +287,7 @@ export class ChipSet {
             disabled: this.disabled || this.readonly,
             'mdc-text-field--with-trailing-icon': true,
         };
+
```
Comment (adrianschmidt) [inline src/components/chip-set/chip-set.tsx:290]: Please avoid whitespace changes in commits with functional changes. I'm guessing this just happened, but if you meant to include it, please put it in a separate commit.

File: src/components/chip-set/chip-set.tsx (hunk 2 above)
Comment (Kiarokh) [reply src/components/chip-set/chip-set.tsx]: Thanks :)

Comment (Kiarokh) [reply src/components/chip-set/chip-set.tsx]: Is it approved then? Should I squash and push?

Comment (adrianschmidt) [reply src/components/chip-set/chip-set.tsx]: I know you saw the comment about whitespace. But yes, you can squash it. I'll approve when you've pushed the squashed commits.

Comment (Kiarokh) [reply src/components/chip-set/chip-set.tsx]: Ah OK. I'll remove the whitespace in the same time then :) I thought it was fine to leave it

Comment (Kiarokh) [reply src/components/chip-set/chip-set.tsx]: Done ✅

**Review by adrianschmidt — APPROVED** [review]
